lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -139,7 +139,11 @@ class window.Turbolinks
         if turbohead.hasAssetConflicts()
           return Turbolinks.fullPageNavigate(url.absolute)
         reflectNewUrl url if options.updatePushState
-        turbohead.insertNewAssets(-> updateBody(upstreamDocument, xhr, options))
+        turbohead.waitForAssets().then(->
+          updateBody(upstreamDocument, xhr, options)
+        ).catch(->
+          # A new request appeared during asset download.
```

Comment:
Is this abusing promises?  Does a request being superseded really count as a rejection?

An alternative to catching would be resolving with an `isCancelled` flag, and only calling `updateBody` if that's `false`.